tests/unit/test_bucket.py

```diff
@@ -0,0 +1,40 @@
+import pytest
+from boto3.exceptions import S3UploadFailedError
+from botocore.exceptions import ClientError
+from moto import mock_s3
+
+from IIIFingest.bucket import upload_directory, upload_image_get_metadata
+
+s3_path = "testing/"
+
+
+@mock_s3
+class TestBucket:
+    test_bucket_name = 'iiif-ingest-test-bucket'
+    file_name = "27.586.1-cm-2016-02-09.tif"
+    key = f"{s3_path}{file_name}"
+
+    def test_upload_image_get_metadata(self, test_images, boto_session):
+        # We need to create the bucket since this is all in Moto's 'virtual' AWS account
+        image_path = test_images[self.file_name]["filepath"]
+        boto_session.resource('s3').create_bucket(Bucket=self.test_bucket_name)
+        image_metadata = upload_image_get_metadata(
+            image_path, self.test_bucket_name, s3_path
+        )
+        assert image_metadata == self.key
+
+    def test_fail_upload_image_get_metadata(self, test_images):
```

Comment:
Looks like this fails because we didn't create the image bucket first for the moto environment? It's a little unclear though -- I had to read all these tests in order and work that out for yourself. (If it's supposed to fail for some other reason, then it's also unclear.) Ditto for `test_upload_directory()` -- I needed to have read `test_upload_image_get_metadata()` to understand the `create_bucket()` call on line 32. 

One approach that might minimize spreading comments around would moving the bucket creation to the class level, and have the "fail" tests use a different bucket name entirely (like `bucket_does_not_exist` or something). That way understanding a test only requires reading it and the standard test setup that precedes it (e.g. class-level and pytest/conftest.py-level).

Reply:
We don't have a use case for creating buckets - that provisioning process is handled by LTS. The IAM roles are pretty tightly scoped and some other actions (eg deleting resources) are also restricted

Reply:
@ColeDCrawford -- I see, that makes sense -- my comment was more about how easy it is to understand the tests in isolation, and the fact that the bucket needs to be setup in moto (cf [this comment on line 18](https://github.com/Harvard-ATG/lts-iiif-ingest-service/pull/10/files/9bfca9d30a7eb88711c8338436fe4601f54ac811#diff-571ecb1c4cac7555b9532d17fb05708bb65cb51c94c3bc4965c75be35e51991aR18)): "We need to create the bucket since this is all in Moto's 'virtual' AWS account"), which is a test requirement but not an application requirement. 

The suggestion would be that writing the tests differently would make the division of the app requirement (bucket existing) and test requirement (provisioning the bucket for the app before testing it) clearer and ensure the same comments don't have to appear in multiple places. I could still be misunderstanding the tests and the app, though! :) Just a thought, in case it's helpful.

Reply:
Got it - yes, doing all the config for the test setup once and in one place would definitely make things clearer, thanks @elliottyates